packages/contractkit/src/wrappers/Governance.ts

```diff
@@ -72,4 +136,298 @@ export class GovernanceWrapper extends BaseWrapper<Governance> {
       stageDurations: res[4],
     }
   }
+
+  getProposalMetadata: (proposalID: BigNumber) => Promise<ProposalMetadata> = proxyCall(
+    this.contract.methods.getProposal,
+    tupleParser(parseNumber),
+    (res) => ({
+      proposer: res[0],
+      deposit: toBigNumber(res[1]),
+      timestamp: toBigNumber(res[2]),
+      transactionCount: toBigNumber(res[3]).toNumber(),
+    })
+  )
+
+  getProposalTransaction: (
+    proposalID: BigNumber,
+    txIndex: number
+  ) => Promise<Transaction> = proxyCall(
+    this.contract.methods.getProposalTransaction,
+    tupleParser(parseNumber, parseNumber),
+    (res) => ({
+      value: toBigNumber(res[0]),
+      destination: res[1],
+      // @ts-ignore string[] bytes type
+      data: toBuffer(res[2]),
+    })
+  )
+
+  isApproved: (proposalID: BigNumber) => Promise<boolean> = proxyCall(
+    this.contract.methods.isApproved, 
+    tupleParser(parseNumber), 
+    identity
+  )
+
+  getApprover = proxyCall(this.contract.methods.approver, undefined, identity)
+
+  async getProposal(proposalID: BigNumber): Promise<Proposal> {
+    const metadata = await this.getProposalMetadata(proposalID)
+    const txIndices = Array.from(Array(metadata.transactionCount).keys())
+    const transactions = await mapAsync(txIndices, (txIndex) =>
+      this.getProposalTransaction(proposalID, txIndex)
+    )
+
+    return {
+      metadata,
+      transactions,
+    }
+  }
+
+  private getTransactionsEncoded(transactions: Transaction[]): TransactionsEncoded {
+    return {
+      values: transactions.map((tx) => tx.value.toString()),
+      destinations: transactions.map((tx) => tx.destination),
+      data: Buffer.concat(transactions.map((tx) => tx.data)),
+      dataLengths: transactions.map((tx) => tx.data.length),
+    }
+  }
+
+  getTransactionsHash(transactions: Transaction[]): Buffer {
+    const encoded = this.getTransactionsEncoded(transactions)
+    return keccak256(
+      this.kit.web3.eth.abi.encodeParameters(
+        ['uint256[]', 'address[]', 'bytes', 'uint256[]'],
+        [encoded.values, encoded.destinations, encoded.data, encoded.dataLengths]
+      )
+    ) as Buffer
+  }
+
+  buildTransactionsFromJSON = (jsonTransactions: JSONTransaction[]) =>
```

Comment:
this can work.. but doesn't provide much type safety; since it's highly dynamic.
When thinking about proposal, my idea, was that we can add a method to `CeloTransaction` to "obtain" either the tx json... or convert it in a proposal.

or maybe have a proposal builder where we add CeloTransactions.

wdyt?